src/middleware/expressLogger.js

```diff
@@ -0,0 +1,34 @@
+const bunyan = require('bunyan')
+const Logger = require('express-bunyan-logger')
+
+const expressLogger = Logger({
+  name: 'expressLogger',
+  parseUA: false,
+  includesFn: (req, res) => {
+    return {
+      origin: req.get('origin'),
+    }
```

Comment:
how this function work?  does this add following vals (origin) to the logged object, and the complete object is the one you showed in comment?

Reply:
Yeah, what it does is attach the `req` and `res` of a request as a top-level property on the log entry, and pass each of them through a serializer. It then adds some other top-level properties based on the request/response, such as a breakdown of the user agent, body, response time, etc.

The `includesFn` option lets you add additional top-level properties. I used it here because I wanted to filter out the large `req` and `res` properties, and only keep certain info at the top level. The original log entry is below - you can see it already had the origin, but as part of the `req` property.

```
{
  "name": "expressLogger",
  "hostname": "42318bdc4c96",
  "pid": 45,
  "req_id": "dabb9bd1-b9e3-455c-be04-400025a7652a",
  "level": 50,
  "remote-address": "::ffff:172.27.0.1",
  "ip": "::ffff:172.27.0.1",
  "method": "GET",
  "url": "/profile?did=did:muport:Qmcz3e72Da5HTb5ahgoCPTVLEHmYF3KavP67TtnmVSeavX",
  "referer": "-",
  "user-agent": {
    "family": "Other",
    "major": "0",
    "minor": "0",
    "patch": "0",
    "device": {
      "family": "Other",
      "major": "0",
      "minor": "0",
      "patch": "0"
    },
    "os": {
      "family": "Other",
      "major": "0",
      "minor": "0",
      "patch": "0"
    }
  },
  "body": {},
  "short-body": "{}",
  "http-version": "1.1",
  "response-time": 241.533212,
  "response-hrtime": [
    0,
    241533212
  ],
  "status-code": 500,
  "req-headers": {
    "user-agent": "PostmanRuntime/7.20.1",
    "accept": "*/*",
    "cache-control": "no-cache",
    "postman-token": "a149b7aa-84f7-437b-8ba8-a5d6a99f8140",
    "host": "localhost:8082",
    "accept-encoding": "gzip, deflate",
    "connection": "keep-alive"
  },
  "res-headers": {
    "x-powered-by": "Express",
    "access-control-allow-origin": "*",
    "content-type": "text/html; charset=utf-8",
    "content-length": "29",
    "etag": "W/\"1d-En0x4ACHDJxoMhDUAf2FNSpBp7I\""
  },
  "req": {
    "method": "GET",
    "url": "/profile?did=did:muport:Qmcz3e72Da5HTb5ahgoCPTVLEHmYF3KavP67TtnmVSeavX",
    "headers": {
      "user-agent": "PostmanRuntime/7.20.1",
      "accept": "*/*",
      "cache-control": "no-cache",
      "postman-token": "a149b7aa-84f7-437b-8ba8-a5d6a99f8140",
      "host": "localhost:8082",
      "accept-encoding": "gzip, deflate",
      "connection": "keep-alive"
    },
    "remoteAddress": "::ffff:172.27.0.1",
    "remotePort": 35406
  },
  "res": {
    "statusCode": 500,
    "header": "HTTP/1.1 500 Internal Server Error\r\nX-Powered-By: Express\r\nAccess-Control-Allow-Origin: *\r\nContent-Type: text/html; charset=utf-8\r\nContent-Length: 29\r\nETag: W/\"1d-En0x4ACHDJxoMhDUAf2FNSpBp7I\"\r\nDate: Wed, 18 Dec 2019 22:04:16 GMT\r\nConnection: keep-alive\r\n\r\n"
  },
  "incoming": "<--",
  "msg": "::ffff:172.27.0.1 <-- GET /profile?did=did:muport:Qmcz3e72Da5HTb5ahgoCPTVLEHmYF3KavP67TtnmVSeavX HTTP/1.1 500 29 - Other 0.0 Other 0.0.0 241.533212 ms",
  "time": "2019-12-18T22:04:16.610Z",
  "v": 0
}
```

Reply:
perfect, thanks for explaining, makes sense!